index.js

```diff
@@ -18,6 +32,8 @@ function isAgent(v) {
  * @api public
  */
 function Agent(callback, _opts) {
+  _opts = urlInstanceToObject(_opts);
```

Comment:
I'd rather not reassign to the parameter. Can we just do `const opts = urlInstanceToObject(_opts);` here? That should also remove line 47 below.